src/widgets/allergies/allergy-record.component.tsx

```diff
@@ -1,165 +1,144 @@
-                  {allergy.reactions
-                    .map(rxn => rxn.reaction.display)
-                    .join(", ")}
-                </td>
-                {/* Onset date is not yet available from the API response */}
-                <td data-testid="onset-date">-</td>
-              </tr>
-            </tbody>
-          </table>
-          {allergy.comment && (
-            <table className={styles.allergyTable}>
-              <thead className="omrs-type-body-regular">
-                <tr>
-                  <th>Comments</th>
-                </tr>
-              </thead>
-              <tbody>
-                <tr data-testid="comment">
-                  <td>{allergy.comment}</td>
-                </tr>
-              </tbody>
-            </table>
-          )}
-          <div className={styles.allergyFooter}></div>
-        </div>
-      </SummaryCard>
-    );
-  }
-
-  function displayDetails() {
-    return (
-      <SummaryCard
-        name="Details"
-        styles={{
-          width: "100%",
-          background: "var(--omrs-color-bg-medium-contrast)"
-        }}
-      >
-        <div className={styles.allergyCard}>
-          <table className={`${styles.allergyTable} ${styles.allergyDetails}`}>
-            <thead className="omrs-type-body-regular">
-              <tr>
-                <th>Last updated</th>
-                <th>Last updated by</th>
-                <th>Last updated location</th>
-              </tr>
-            </thead>
-            <tbody>
-              <tr>
-                <td data-testid="last-updated">
-                  {allergy.auditInfo.dateChanged
-                    ? dayjs(allergy.auditInfo.dateChanged).format("DD-MMM-YYYY")
-                    : "-"}
-                </td>
-                <td data-testid="updated-by">
-                  {allergy.auditInfo.creator.display}
-                </td>
-                <td data-testid="update-location">-</td>
-              </tr>
-            </tbody>
-          </table>
-        </div>
-      </SummaryCard>
-    );
-  }
+  }, [isLoadingPatient, patient, match.params]);
 
   return (
     <>
-      {allergy && (
-        <div className={styles.allergySummary}>{displayAllergy()}</div>
-      )}
-      {allergy && (
-        <div className={styles.allergySummary}>{displayDetails()}</div>
+      {!!(allergy && Object.entries(allergy).length) && (
+        <div className={styles.allergyContainer}>
+          <SummaryCard
+            name="Allergy"
+            styles={{ width: "100%" }}
+            editComponent={AllergyForm}
+            showComponent={() =>
+              openWorkspaceTab(AllergyForm, "Edit Allergy", {
+                allergyUuid: allergy.id
+              })
+            }
+            link={`/patient/${patientUuid}/chart/allergies`}
+          >
+            <div
+              className={`omrs-type-body-regular ${styles.allergyCard} ${
+                allergy.reactionSeverity === "Severe"
```

Comment:
Inconsistent. Below we always use `allergy?.` as if this may be undefined, but here we directly use it.

I would prefer always direct usage with only a single check upfront. We already have this check upfront (`!!allergy`).